examples/dht_node_new.rs

```diff
@@ -0,0 +1,109 @@
+/*
+    Copyright (C) 2013 Tox project All Rights Reserved.
+    Copyright © 2017 Zetok Zalbavar <zexavexxe@gmail.com>
+    Copyright © 2018 Namsoo CHO <nscho66@gmail.com>
+
+    This file is part of Tox.
+
+    Tox is libre software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Tox is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with Tox.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+// an example of DHT node with current code
+//
+extern crate tox;
+extern crate futures;
+extern crate tokio_core;
+extern crate tokio_io;
+extern crate tokio_timer;
+
+#[macro_use]
+extern crate log;
+extern crate env_logger;
+
+use futures::*;
+use futures::sync::mpsc;
+use tokio_core::reactor::Core;
+use tokio_core::net::UdpSocket;
+//use tokio_timer;
+
+use std::cell::RefCell;
+use std::net::SocketAddr;
+use std::rc::Rc;
+use std::io::{ErrorKind, Error};
+use std::time;
+
+use tox::toxcore::dht_new::codec::*;
+use tox::toxcore::dht_new::server::*;
+
+fn main() {
+    env_logger::init();
+
+    let local: SocketAddr = "127.0.0.1:12345".parse().unwrap();
+
+    let mut core = Core::new().unwrap();
+    let handle = core.handle();
+
+    // Bind a UDP listener to the socket address.
+    let listener = UdpSocket::bind(&local, &handle).unwrap();
+
+    let node = Rc::new(RefCell::new(Server::new().unwrap()));
+
+    let (sink, stream) = listener.framed(DhtCodec).split();
+    // Create a channel for this socket
+    let (tx, rx) = mpsc::unbounded::<DhtUdpPacket>();
+    // The server task asynchronously iterates over and processes each
+    // incoming packet.
+    let handler = stream.for_each(move |(addr, packet)| {
+        let tx_c = tx.clone();
+        let pk = Server::get_pk(&packet.clone().unwrap());
+        let client = node.borrow_mut().create_client(&addr, pk, tx_c);
```

Comment:
It seems to me that server should take only packet and maybe socket address but not the whole client. It should manage clients inside so the user of server api wouldn't even know about it.